client/packages/invoices/src/InboundShipment/DetailView/modals/Donor/DonorEditModal.tsx

```diff
@@ -0,0 +1,202 @@
+import React, { useEffect, useState } from 'react';
+import {
+  Autocomplete,
+  Box,
+  useDialog,
+  DialogButton,
+  useNotification,
+  useTranslation,
+  InputWithLabelRow,
+  UpdateDonorMethodInput,
+  Typography,
+  LocaleKey,
+  useConfirmationModal,
+} from '@openmsupply-client/common';
+import { useName } from 'packages/system/src';
+import { getDonorUpdateKeys } from 'packages/system/src/utils';
+import { useInbound } from '../../../api';
+
+interface DonorEditModalProps {
+  invoiceId: string;
+  donorId: string;
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+interface DonorOption {
+  label: string;
+  value: string | null;
+  id: string | null;
+}
+
+interface MethodOption {
+  label: string;
+  value: UpdateDonorMethodInput;
+  message: LocaleKey;
+}
+
+export const DonorEditModal = ({
+  invoiceId,
+  donorId,
+  isOpen,
+  onClose,
+}: DonorEditModalProps) => {
+  const { error } = useNotification();
+  const t = useTranslation();
+  const { Modal } = useDialog({ isOpen, onClose });
+  const { mutateAsync } = useInbound.document.update();
+  const defaultMethod = {
+    label: t(getDonorUpdateKeys(UpdateDonorMethodInput.None).label),
+    value: UpdateDonorMethodInput.None,
+    message: getDonorUpdateKeys(UpdateDonorMethodInput.None).message,
+  };
+
+  const [val, setVal] = useState<DonorOption | undefined>();
+  const [method, setMethod] = useState<MethodOption>(defaultMethod);
+
+  const width = '350px';
+
+  const getConfirmation = useConfirmationModal({
+    title: t('heading.are-you-sure'),
+    message: t('messages.update-all-children-lines'),
+  });
+
+  const { data } = useName.document.donors();
+
+  const donorOptions: DonorOption[] =
+    data?.nodes.map(node => ({
+      label: node.name,
+      value: node.id,
+      id: node.id,
+    })) ?? [];
+
+  const nullOption = {
+    label: t('label.set-to-none'),
+    value: null,
+    id: null,
+  };
+
+  donorOptions.push(nullOption);
+
+  type IsMethod = keyof typeof UpdateDonorMethodInput;
+  const methodOptions: MethodOption[] = (
+    Object.keys(UpdateDonorMethodInput) as Array<IsMethod>
+  ).map(key => {
+    const valueKeys = getDonorUpdateKeys(UpdateDonorMethodInput[key]);
+    return {
+      label: t(valueKeys.label),
+      value: UpdateDonorMethodInput[key],
+      message: valueKeys.message,
+    };
+  });
+
+  useEffect(() => {
+    const donor = donorOptions?.find(donor => (donor.id = donorId));
+    donor ? setVal(donor) : setVal(nullOption);
+  }, [donorId, data]);
+
+  const confirm = () => {
+    mutateAsync({
+      id: invoiceId,
+      defaultDonorId: val?.id,
+      updateDonorMethod: method.value,
+    });
+  };
+
+  const handleSave = () => {
+    // only prompt confirm if we are making updates to children lines
+    method.value != UpdateDonorMethodInput.None
+      ? getConfirmation({
+          onConfirm: () => {
+            confirm();
+          },
+        })
+      : confirm();
```

Comment:
Here am only showing the confirmation modal when we are actually changing lines. Otherwise I figured it wouldn't be warranted.